### DISCUSSION TITLE
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @ai16z/agent@0.1.6-alpha.4 start: `node --loader ts-node/esm src/index.ts "--isRoot"`

### DISCUSSION THREAD

**Discussion by fadai88** [elizaOS/eliza#1280] [General]
When I run "sh scripts/start.sh" I get a lot of downloads, installs etc, and at the end this message:

ℹ INFORMATIONS
   Loading embedding settings: 
   {"USE_OPENAI_EMBEDDING":"","USE_OLLAMA_EMBEDDING":"","OLLAMA_EMBEDDING_MODEL":"mxbai-embed-large"} 

 ℹ INFORMATIONS
   Loading character settings: 
   {"ARGV":["/home/fmiren/.nvm/versions/node/v20.0.0/bin/node","/home/fmiren/eliza/agent/src/index.ts","--isRoot"],"CWD":"/home/fmiren/eliza/agent"} 

Loaded .env file from: /home/fmiren/eliza/.env
 ℹ INFORMATIONS
   Parsed settings: 
   {"USE_OPENAI_EMBEDDING":"","USE_OPENAI_EMBEDDING_TYPE":"string","USE_OLLAMA_EMBEDDING":"","USE_OLLAMA_EMBEDDING_TYPE":"string","OLLAMA_EMBEDDING_MODEL":"mxbai-embed-large"} 

Killed
/home/fmiren/eliza/agent:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @ai16z/agent@0.1.6-alpha.4 start: `node --loader ts-node/esm src/index.ts "--isRoot"`
Exit status 137
 ELIFECYCLE  Command failed with exit code 137.
-e Failed to start project

What does it mean? Why does it fail? Is it ok or not?

**Comment by lorcann-rauzduel** [comment]
Same here

**Comment by gepng** [comment]
Restarting from scratch using the docs fixed it for me. But before doing that you also have to make sure that your node version is exactly `23.3.0`

**Reply by patrickjsimons** [reply]
Yep, I see v20.0.0 in the console output above. 

For others running into similar problems:

1) download and use node version manager 'nvm' to setup the correct node environment. After installing nvm run the following command each time you open a new terminal as your default node version might be set to something else:

 nvm use 23.3.0 

2) Make sure you initially checkout a suitable version from the main branch i.e. don't checkout a branch like 'develop'  

 git clone https://github.com/elizaOS/eliza.git
 cd eliza
 git checkout $(git describe --tags --abbrev=0)

3) After switching between branches it's best to clean the project and install all the related dependencies that are listed in package.json 

 pnpm clean
 pnpm install --no-frozen-lockfile

4) Build and start the project:
 
 pnpm build
 pnpm start --character="yourCharacter.json" 

PS - the official documentation is a great reference point!

**Comment by Belief208** [comment]
same here

**Comment by y4my4my4m** [comment]
same issue, on arch, node v23.3.0
